### DISCUSSION TITLE
Replace in files

### DISCUSSION THREAD

**Discussion by davidhealey** [orbitalquark/textadept#140] [Q&A] [answered]
I know it's possible to search for text within a folder of files, but is it also possible to replace the text in those files?

**Comment by orbitalquark** [comment, marked as answer]
No, sorry, this has to be done manually.

**Reply by davidhealey** [reply]
Think this feature will ever be added?

**Comment by orbitalquark** [comment]
No, I don't think so. This feature can be automated via macro or Lua script. For example:

1. Ctrl+Shift+F and perform "find in files".
2. Ctrl+F to show "Replace" field.
3. Enter text to replace. Press Esc to hide the pane.
4. F9 to start recording macro.
5. Ctrl+Alt+G to jump to next result.
6. Ctrl+Alt+R to replace.
7. F9 to stop recording macro.
8. Shift+F9 to repeat 5-7 for each result.

I personally do not trust a blind "Replace All" in an "In Files" search.

**Reply by davidhealey** [reply]
Thanks I'll give it a try and play around with a Lua implementation as well.
